manticore/core/manticore.py

```diff
@@ -1065,18 +1082,33 @@ def run(self):
         # different process modifies the state
         self.stcache = weakref.WeakValueDictionary()
 
+        if self._introspector is None:
+            self._introspector = self._introspection_plugin()
+            self.register_plugin(self._introspector)
+
         # Lazy process start. At the first run() the workers are not forked.
         # This actually starts the worker procs/threads
         if self.subscribe:
             # User subscription to events is disabled from now on
             self.subscribe = None
 
+        # Passing generators to callbacks is a bit hairy because the first callback would drain it if we didn't
+        # clone the iterator in event.py. We're preserving the old API here, but it's something to avoid in the future.
         self._publish("will_run", self.ready_states)
         self._running.value = True
         # start all the workers!
         for w in self._workers:
             w.start()
 
+        # Create each daemon thread and pass it `self`
+        if not self._daemon_threads:  # Don't recreate the threads if we call run multiple times
```

Comment:
So we have this here just because we need the workers to start before this threads?
(As opposed to have a Thread started from the main script before the run)

If the workers were stated at `Manticore.__init__()` could not we do something like the following from main():  
```
with SomeDaemonThread():
    m.run()
```
?

Reply:
To be honest, I didn't think very hard about where to start the daemon threads. In my mind, it doesn't make much sense to have them running until Manticore is also running, but I don't think there's necessarily any problem with starting them in `__init__`.

Reply:
Oh..I was confused. I thought for how it looks like that it was put after the worker.start() on purpose. 
Also it happens that a fork will not fork/copy those threads to the forked processes anyway.  :) 

So then we could in fact start the daemon threads from the main script? 
```
m = Manticore()
with SomeDaemonThread():
    m.run()
```
I still need to dig out the reason why this could be a ManticoreBase responsibility.

Reply:
> Also it happens that a fork will not fork/copy those threads to the forked processes anyway 

Right, there should only ever be one copy of those threads running in the main process. 

> So then we could in fact start the daemon threads from the main script?

I think the context manager approach could work, but there are a few things to work out. The nice thing about it is that it would make it easy to notify daemon threads that a run had completed. The somewhat difficult aspect is that for things like the TU, which needs to capture log messagesI, we don't want the thread to exit until the entire program exits - not just when `run` completes. So you could still start the thread via a context manager, but it'd have to live on in the background somewhere.